Disaster-Response-Platform/backend/Controllers/uprofile_SocMed_controller.py

```diff
@@ -16,14 +16,23 @@
     status.HTTP_404_NOT_FOUND: {"model": Error},
     status.HTTP_401_UNAUTHORIZED: {"model": Error}
 })
-async def get_user_socialmedia(response: Response, anyuser:str= None, platform_name:str = None, current_username: str = Depends(authentication_service.get_current_username)):
+async def get_user_socialmedia(response: Response, platform_name:str = None, anyuser:str= None,
+                               current_username: str = Depends(authentication_service.get_current_username)):
     if anyuser is None:
         if platform_name is None:
             username = current_username
         else:
-            username = None
+            if (authentication_service.is_admin(current_username)):
+                username = None
+            else:
+                response.status_code = HTTPStatus.UNAUTHORIZED
+                return create_json_for_error("Action prohibited", "Only users with ADMIN role can do that")
```

Comment:
Status should be status.HTTP_403_FORBIDDEN, since the user is authorized, but does not have access.